src/app/input-bar/input-bar.component.ts

```diff
@@ -18,25 +19,167 @@ export class InputBarComponent implements OnInit {
+    // We need to scan filters for regex so we can highlight matches
+    const newRegexArr: RegExp[] = [];
+    options.filters.forEach((filter: FilterOption<any>) => {
+      if (filter.filterKey === FilterKeys.Regex && !filter.negation) {
+        newRegexArr.push(new RegExp(`(${filter.inputString})`, 'g'));
+      }
+    });
+    this.ccdService.updateRegex(newRegexArr);
 
   }
 
-  parseInput(input: string) {
-    try {
-      return JSON.parse(input);
-    } catch (e) {
-      // By deault we assume user is just inputting a regex
-      return {regex: input};
+  private parseInput(input: string) {
+    const filters: FilterOption<any>[] = [];
+    let groupBy: GroupByKeys = GroupByKeys.None;
+
+    if (!input) {
+      return {
+        filters,
+        groupBy
+      };
+    }
+    const splitInput = input.trim().split(' ');
+
+    if (splitInput.length === 1) {
+      const newFilter = this.parseArgument(splitInput[0], true);
+      if (newFilter) {
+        filters.push(newFilter);
+      }
+    } else {
+      for (let i = 0; i < splitInput.length; i++) {
+        if (splitInput[i] !== '{') {
+          const newFilter = this.parseArgument(splitInput[i], false);
+          if (newFilter.validInput) {
+            filters.push(newFilter);
+          }
+        } else if (i + 2 >= splitInput.length || splitInput[i + 2] !== '}') {
+          // To group we require the syntax '{ grouping_key }'
+          console.error('Invalid input, to group links make sure there are ' +
+              'spaces between the brackets and the grouping key.');
+          break;
+        } else {
+          for (const key in GroupByKeys) {
+            if (GroupByKeys[key] === splitInput[i + 1]) {
+              groupBy = GroupByKeys[key];
+              break;
+            }
+          }
+          i += 2;
+        }
+      }
     }
+
+    return {
+      filters,
+      groupBy
+    };
   }
+
+  /**
+   * Parses a full filter argument, parses it and constructs a FilterOption
+   * object filled with the relevant data.
+   *
+   * @param { string } str The full filter argument as a string
+   * @param { boolean } onlyArg Whether it's the only argument and we should
+   * accept it as a regex filter if there is no modifier
+   * @returns { FilterOption<any> } The constructed FilterOption object
+   */
+  private parseArgument(str: string, onlyArg: boolean): FilterOption<any> {
+    // By default we assume a regex unless we find a modifier
+    const filterOption = {
+      filterKey: undefined,
+      inputString: str,
+      value: undefined,
+      negation: false,
+      validInput: true
+    };
+
+    this.parseNegation(filterOption);
+
+    for (const key in FilterKeys) {
+      if (filterOption.inputString.startsWith(FilterKeys[key])) {
+        this.parseModifier(filterOption, FilterKeys[key]);
+        break;
+      }
+    }
+
+    // If the argument had no valid modifier
+    if (!filterOption.filterKey) {
+      if (onlyArg) {
```

Comment:
What do you think about treating all inputs without modifier as regex? I think it would be more easier to handle (you don't need to have onlyArg argument at all)